detect/archive.go

```diff
@@ -0,0 +1,96 @@
+package detect
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/archives"
+	"github.com/zricethezav/gitleaks/v8/sources"
+)
+
+// IsArchive asks archives.Identify (with a nil stream, so only the filename)
+// whether this file would be handled by an Extractor. If Identify returns
+// a Format implementing archives.Extractor, we treat it as an archive.
+func IsArchive(path string) bool {
+	format, _, err := archives.Identify(context.Background(), path, nil)
+	if err != nil {
+		// no matching format at all
+		return false
+	}
+	_, ok := format.(archives.Extractor)
+	return ok
+}
+
+// ExtractArchive extracts all files from archivePath into a temp dir.
+// Returns the list of ScanTargets (with real file paths) and the temp dir for cleanup.
+func ExtractArchive(archivePath string) ([]sources.ScanTarget, string, error) {
+	tmpDir, err := os.MkdirTemp("", "gitleaks-archive-")
+	if err != nil {
+		return nil, "", err
+	}
+
+	f, err := os.Open(archivePath)
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		return nil, "", err
+	}
+	defer f.Close()
+
+	ctx := context.Background()
```

Comment:
If I push a PR adding top-level context propagation, would you merge it?

Reply:
assuming it didn't break anything, sure